src/transactions/transactions.util.ts

```diff
@@ -53,11 +69,35 @@ export async function processTransaction<
 >(
   method: Method<MethodArgs, ReturnType, TransformedReturnType>,
   args: MethodArgs,
-  opts: ProcedureOpts
+  opts: ProcedureOpts,
+  dryRun = false
 ): Promise<TransactionResult<TransformedReturnType>> {
   try {
     const procedure = await prepareProcedure(method, args, opts);
-    const result = await procedure.run();
+    const {
+      fees,
+      payingAccountData: { balance, type, account },
+    } = await procedure.getTotalFees();
+    const supportsSubsidy = await procedure.supportsSubsidy();
+
+    const details: TransactionDetails = {
+      status: procedure.status,
+      fees,
+      supportsSubsidy,
+      payingAccount: {
+        balance,
+        type,
+        address: account.address,
+      },
+    };
+
+    let result = undefined as unknown as TransformedReturnType;
```

Comment:
This assertion is dangerous

Can't `TransformedReturnType` accept an undefined value for result instead?